opentreemap/stormwater/migrations/0007_drainage_area_permissions.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def add_permissions_all(apps, schema_editor):
+    Instance = apps.get_model('treemap', 'Instance')
+    Role = apps.get_model('treemap', 'Role')
+    FieldPermission = apps.get_model('treemap', 'FieldPermission')
+    need_to_clear_cache = False
+
+    def update_field_permissions(perm_specs, role, instance):
+        existing = FieldPermission.objects.filter(
+            role=role, instance=instance)
+        if existing.exists():
+            for perm in perm_specs:
+                perm['defaults'] = {
+                    'permission_level': role.default_permission}
+                FieldPermission.objects.get_or_create(**perm)
+            return False
+        else:
+            perm_specs = [FieldPermission(**perm) for perm in perm_specs]
+            for perm in perm_specs:
+                perm.permission_level = role.default_permission
+            FieldPermission.objects.bulk_create(perm_specs)
+            return True
+
+    for instance in Instance.objects.filter(config__contains='"Plot",'):
+        types = instance.config['map_feature_types']
+        if 'Bioswale' in types or 'RainGarden' in types:
+            for role in Role.objects.filter(instance=instance):
+                perm_specs = [
+                    {'model_name': 'Bioswale',
+                     'field_name': 'drainage_area',
+                     'role': role,
+                     'instance': instance},
+                    {'model_name': 'RainGarden',
+                     'field_name': 'drainage_area',
+                     'role': role,
+                     'instance': instance}]
+                need_to_clear_cache = \
+                    update_field_permissions(perm_specs, role, instance) or \
+                    need_to_clear_cache
+
+    if need_to_clear_cache:
+        from object_caches import clear_caches
+        clear_caches()
```

Comment:
`clear_caches()` is not the method we want to call here - it will clear the in-memory cache, but will not update the cache key.  The method we need to call to do that is [`increment_adjuncts_timestamp`](https://github.com/OpenTreeMap/otm-core/blob/develop/opentreemap/treemap/lib/object_caches.py#L63-L72) (though we can't actually call it directly, because we need to use migration models...), and by incrementing it, the next attempt to access the cache will clear the cache for us.

Instead, we should update the cache key on a per instance basis:
```python
for instance in Instance.objects.filter(config__contains='"Plot",'):
    need_to_clear_cache = False
    types = instance.config['map_feature_types']
        if 'Bioswale' in types or 'RainGarden' in types:
            for role in Role.objects.filter(instance=instance):
                ...
                ...
                if update_field_permissions(perm_specs, role, instance):
                    need_to_clear_cache = True

    if need_to_clear_cache:
        # Copied from object_caches.increment_adjuncts_timestamp
        Instance.objects.filter(pk=instance.id) \
            .update(adjuncts_timestamp=F('adjuncts_timestamp') + 1)
```